test/tools/evmoneRunner.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/evmoneRunner.h>
+#include <iostream>
+#include <assert.h>
+#include <fstream>
+#include <limits>
+
+using namespace external::evmone;
+using namespace std;
+
+evmc_host_interface EvmOneVM::interface{
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		[](evmc_context* ctx) { return static_cast<EvmOneVM*>(ctx)->tx_context; },
+		{},
+		{},
+};
+
+void EvmOneVM::execute(const evmc_message& _m, std::basic_string<uint8_t> _code)
```

Comment:
```suggestion
void EvmOneVM::execute(evmc_message const& _m, std::basic_string<uint8_t> _code)
```

Reply:
Is it really `std::basic_string<uint8_t>` instead of `std::string`?

Reply:
Yes, please see 

https://github.com/chfast/evmone/blob/15c67f8f6dcfa0a0f98b04f97b87c7501351cca0/test/unittests/execution_fixture.hpp#L79

and

https://github.com/chfast/evmone/blob/7ef24ab4551f1a5658090138de4a12e0888cae50/test/utils/utils.hpp#L12

Reply:
Oh! Does this need c++17?

Reply:
evmone needs c++17 but I assumed if we replace `string_view` with `string` we won't need c++17. At least that's the hope.

Reply:
I don't think replacing it like that will work, because `string` always performs a copy and thus has refers to a different object.

Reply:
And even if we have evmc is a binary library, linking against it might be problematic because it needs a newer version of libstdc++

Reply:
you're right. I hadn't considered implications of c++17 dependency on linking.

Reply:
This is list of undefined symbols in `libevmone.so`:
```
U __cxa_begin_catch@@CXXABI_1.3
                 U __cxa_end_catch@@CXXABI_1.3
                 w __cxa_finalize@@GLIBC_2.2.5
                 U __cxa_rethrow@@CXXABI_1.3
                 U free@@GLIBC_2.2.5
                 w __gmon_start__
                 U __gxx_personality_v0@@CXXABI_1.3
                 w _ITM_deregisterTMCloneTable
                 w _ITM_registerTMCloneTable
                 U malloc@@GLIBC_2.2.5
                 U memcpy@@GLIBC_2.14
                 U memmove@@GLIBC_2.2.5
                 U memset@@GLIBC_2.2.5
                 U __stack_chk_fail@@GLIBC_2.4
                 U _Unwind_Resume@@GCC_3.0
                 U operator delete(void*)@@GLIBCXX_3.4
                 U operator new(unsigned long)@@GLIBCXX_3.4
                 U std::__throw_bad_alloc()@@GLIBCXX_3.4
                 U std::__throw_length_error(char const*)@@GLIBCXX_3.4
                 U std::terminate()@@GLIBCXX_3.4
                 U vtable for __cxxabiv1::__class_type_info@@CXXABI_1.3
                 U vtable for __cxxabiv1::__si_class_type_info@@CXXABI_1.3
```

Reply:
> I don't think replacing it like that will work, because string always performs a copy and thus has refers to a different object.

Quoting @chfast Using std::string is ok. It will be converted to std::string_view

Reply:
Note to self: https://stackoverflow.com/questions/46746878/is-it-safe-to-link-c17-c14-and-c11-objects